<a href="https://colab.research.google.com/github/DIAPkuik237/regression-multiple-neurone/blob/master/De_la_r%C3%A9gression_simple_au_neurone_artificiel.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
# Régression linéaire multiple + sigmoïde : prédiction du score de l'étudiant D

import numpy as np

# ================================
# Données : Étudiants A, B, C
# ================================
# Variables : [temps d’étude, temps de sommeil]
X = np.array([
    [2, 6],
    [3, 7],
    [1, 8]
])

# Scores correspondants
y = np.array([0.3, 0.5, 0.4])

# ================================
# Ajout du biais (colonne de 1)
# ================================
# X_b = matrice étendue : [1, étude, sommeil]
X_b = np.c_[np.ones((X.shape[0], 1)), X]

# ================================
# Régression linéaire par moindres carrés
# ================================
# Formule : theta = (X^T X)^(-1) X^T y
theta_best = np.linalg.inv(X_b.T @ X_b) @ X_b.T @ y

print("✅ Poids appris (biais, étude, sommeil) :", theta_best)

# ================================
# Prédiction pour l’étudiant D : 4h d’étude, 5h de sommeil
# ================================
x_D = np.array([1, 4, 5])  # [biais, étude, sommeil]
z_D = x_D @ theta_best

print("📊 Score prédit (avant activation) :", z_D)

# ================================
# Ajout de la fonction d’activation sigmoïde
# ================================
def sigmoid(z):
    return 1 / (1 + np.exp(-z))

score_prob = sigmoid(z_D)

print("🔮 Probabilité de réussite (après sigmoïde) :", score_prob)


✅ Poids appris (biais, étude, sommeil) : [-0.5  0.1  0.1]
📊 Score prédit (avant activation) : 0.39999999999999575
🔮 Probabilité de réussite (après sigmoïde) : 0.598687660112451


###✅ Résultat à l’exécution :

 Poids appris (biais, étude, sommeil) : [-0.5  0.1  0.1]

📊 Score prédit (avant activation) : 0.39999999999999575

🔮 Probabilité de réussite (après sigmoïde) : 0.598687660112451

=> les résultats peuvent varier selon les données

##🧠 Interprétation :
Poids appris :

a1 = 0.1 → Chaque heure d’étude augmente le score de 0.1

a2 = 0.1 → Chaque heure de sommeil augmente aussi le score de 0.1

b = -0.5 → Le biais est négatif, il baisse la prédiction de base

Score prédictif brut ≈ 0.4

Après sigmoïde : probabilité ≈ 60%

Le modèle estime donc que l’étudiant D a environ 60 % de chances de réussir, selon ses 4h d’étude et 5h de sommeil.

⚠️ Remarque importante (à intégrer dans ton article)
Tu peux ajouter une note en encadré ou dans la conclusion :

🔎 Remarque :
Les résultats varient en fonction des données utilisées.

Avec plus de données ou des données différentes, les poids et la prédiction changeraient.

C’est là qu’intervient l’apprentissage automatique, où le modèle s’améliore avec plus d’exemples — ce sera le sujet du prochain article !